### PR TITLE
Use `DragAndDropManager` instead of having wrapper

### PR DESCRIPTION
**Pull request by MatkovIvan**
In order to align fork and AOSP, we need to reuse existing common. Make `DragAndDropManager` public under `@InternalComposeUiApi` instead of having an extra wrapper.

### CODE REVIEW

**Comment by igordmn** [comment]
[Here](https://github.com/JetBrains/compose-multiplatform-core/pull/1493#discussion_r1720077380) I suggested an alternative approach, which doesn't expose this class as public.

If we still decide that it is needed as public, first we need to do this in AOSP, and only then cherry-pick it to the fork. But we can keep the PR for discussion.

**First thought** _(see my second thought below, it overrides the first one)_

Also, I am not completely sure that we need to expose such classes, because:
- only UI module needs it, for Foundation we have another exposed public classes. The situation would be different if it was some class that is needed for Foundation and higher levels.
- it isn't needed as public for Android target, because Android target doesn't have "public" Compose core
- we on the other hand have independent Compose core that requires pseudo-public API
- our requirements can be only our requirements here, we shouldn't increase burden for maintaining Android as well.

TLDR. We don't need this public:
```
internal PlatformThing <- UI
```

We only need this public:
```
@Internal PlatformSceneThing <- UI.ComposeScene
public AnotherPlatformThing <- Foundation
```

**Comment by igordmn** [comment]
> burden for maintaining

On second thought, if we agree to expose some needed classes as `public @InternalComposeUiApi`, which shouldn't require additional burden in theory, these kind of changes looks beneficial
